bluez/examples/bluetoothctl.rs

```
@@ -0,0 +1,275 @@
+						continue;
+					}
+					println!(
+						"{} ({})",
+						proxy
+							.device
+							.name()
+							.await
+							.unwrap_or(proxy.device.address().await.unwrap()),
+						proxy
+							.device
+							.inner()
+							.get_property::<String>("Icon")
+							.await
+							.unwrap_or("unknown".to_owned())
+					);
+				}
+			}
+		},
+
+		Some("nearby-devices") => {
+			futures::future::join_all(adapters.iter().map(|adapter| adapter.start_discovery()))
+				.await;
+			match bluez_zbus::get_devices(&connection, None).await {
+				Err(why) => {
+					eprintln!("error: could not get devices: {why}");
+					return Ok(ExitCode::FAILURE);
+				}
+				Ok(devices) => {
+					for proxy in devices {
+						if proxy.device.paired().await? {
+							continue;
+						}
+						println!(
+							"{} ({})",
+							proxy
+								.device
+								.name()
+								.await
+								.unwrap_or(proxy.device.address().await.unwrap()),
+							proxy
+								.device
+								.inner()
+								.get_property::<String>("Icon")
+								.await
+								.unwrap_or("unknown".to_owned())
+						);
+					}
+				}
+			}
+		}
+
+		Some("connect") => match parser.free_from_str::<String>().ok().as_deref() {
+			Some(addr_or_alias) => match bluez_zbus::get_devices(&connection, None).await {
+				Err(why) => {
+					eprintln!("error: could not get devices: {why}");
+					return Ok(ExitCode::FAILURE);
+				}
+				Ok(devices) => {
+					let devices: Vec<bluez_zbus::BluetoothDevice> =
+						futures::future::join_all(devices.into_iter().map(|proxy| async {
+							match (proxy.device.name().await, proxy.device.address().await) {
+								(Ok(alias), _) if alias == addr_or_alias => Some(proxy),
+								(_, Ok(addr)) if addr == addr_or_alias => Some(proxy),
+								_ => None,
+							}
+						}))
+						.await
+						.into_iter()
+						.flatten()
+						.collect();
+					if devices.is_empty() {
+						eprintln!("No device found");
+						return Ok(ExitCode::FAILURE);
```

It may be better to return a custom error type